errorcodes/errorcodes.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errorcodes
+
+import (
+	"errors"
+
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/runtime/protoiface"
+)
+
+const (
+	ErrorCodesFeatureMetadataKey = "error_codes_feature"
```

Comment:
Where would this key appear?

Reply:
So as I understand- we are using this key to change behavior here based on a feature gate in dapr/dapr? I think it would be best for this package to be as simple as possible, and use this package or not based on the presence of the feature gate in dapr/dapr rather than needing to learn the feature gate here as well.

Reply:
The feature flag is here because contrib also needs to know if the feature is enabled. This is not just runtime.

Reply:
I think it makes more sense for all the calling code to be feature gate aware rather than kit. Making kit aware of feature flags means that its functionality is tied to the version of runtime, rather than a separate self contained utility library. It also means there would now be a bi-directional relationship between kit and other repos, where it is only one way currently.

Contrib is already tightly coupled with the runtime version, what would be the implication of having the feature gate aware code moved there instead? Another option is to have a `kit/errors.Builder` struct which is given an `Options`, however I would assume this would end up being more verbose than the caller handling the feature gate logic.

Reply:
Agreed! I'm removing the logic of the feature gate and having it on dapr/dapr + contrib.